### PR TITLE
Add cluster into endpoint workload metadata.

### PR DESCRIPTION
**Pull request by bianpengyuan**
fixes https://github.com/istio/istio/issues/29373

### CODE REVIEW

File: pilot/pkg/xds/ep_filters.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 					},
 				}
 				// TODO: figure out a way to extract locality data from the gateway public endpoints in meshNetworks
-				gwEp.Metadata = util.BuildLbEndpointMetadata(network, model.IstioMutualTLSModeLabel, "", "", labels.Instance{})
+				gwEp.Metadata = util.BuildLbEndpointMetadata(network, model.IstioMutualTLSModeLabel, "", "", "", labels.Instance{})
```
Comment (stevenctl) [inline pilot/pkg/xds/ep_filters.go:122]: b.clusterID should work here.. it's easy to mix that with clusterName which is the envoy cluster.

Comment (bianpengyuan) [reply pilot/pkg/xds/ep_filters.go]: Thanks! Done.

File: pilot/pkg/networking/util/util.go [hunk 2]
```diff
@@ -498,7 +498,7 @@ func MergeAnyWithAny(dst *any.Any, src *any.Any) (*any.Any, error) {
 }
 
 // BuildLbEndpointMetadata adds metadata values to a lb endpoint
-func BuildLbEndpointMetadata(network, tlsMode, workloadname, namespace string, labels labels.Instance) *core.Metadata {
+func BuildLbEndpointMetadata(network, tlsMode, workloadname, namespace, clusterName string, labels labels.Instance) *core.Metadata {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/util/util.go:501]: s/clusterName/clusterID

**Review by hzxuzhonghu — COMMENTED** [review]
Should we also protect the older proxies? Only set the metadata for proxy >=1.10.

File: pilot/pkg/networking/util/util.go (hunk 2 above)
Comment (bianpengyuan) [reply pilot/pkg/networking/util/util.go]: Done.

**Comment by bianpengyuan** [comment]
@hzxuzhonghu this is fine, 1.9 proxy could handle this safely: https://github.com/istio/proxy/blob/release-1.9/extensions/common/context.cc#L301

**Review by howardjohn — APPROVED** [review]

**Pull request merged**